Monika After Story/game/special-effects.rpy

```diff
@@ -1,5 +1,670 @@
+                Nevertheless, I decided to handle this case so we can be sure that isn't our fault.
+            """
+            rv = super(ParallaxSprite, self).__getstate__()
+            rv["_render"] = None
+
+            return rv
+
+        def event(self, ev, x, y, st):
+            """
+            The event handler
+            TODO: allow to move around and zoom in/out with keyboard
+            """
+            if self._enable_events:
+                if ev.type == pygame.MOUSEMOTION:
+                    self.update_mouse_pos()
+                    self.update_offsets()
+
+                elif ev.type == pygame.MOUSEBUTTONDOWN:
+                    if self.min_zoom != self.max_zoom:
+                        if ev.button == 4:
+                            self.zoom += 0.1
+
+                        elif ev.button == 5:
+                            self.zoom -= 0.1
+
+                elif ev.type == pygame.MOUSEBUTTONUP:
+                    if ev.button == 1:
+                        # if self.is_focused():
+                        if self._render is not None and self._render.is_pixel_opaque(x, y):
+                            x_pos, y_pos, x_anchor, y_anchor, x_offset, y_offset, subpixel = self._transform.get_placement()
+                            # FIXME: Support relative xpos and ypos (add float handling)
+                            x_pos = x_pos or 0
+                            y_pos = y_pos or 0
+                            x_offset = x_offset or 0
+                            y_offset = y_offset or 0
+                            return self._transform.event(ev, x - x_pos - x_offset, y - y_pos - y_offset, st)
+
+            return None
+
+        def render(self, width, height, st, at):
+            """
+            The render method
+            """
+            img_surf = renpy.render(self._transform, width, height, st, at)
+            # NOTE: This has to do full size screen render + instead of blit use place
+            render = renpy.Render(width, height)
+            render.place(self._transform, x=0, y=0, render=img_surf)
+            # render.add_focus(self, None, 0, 0, renpy.config.screen_width, renpy.config.screen_height, 0, 0, render)
+            self._render = render
+
+            return render
+
+        def visit(self):
+            """
+            Returns the transform for prediction
+            """
+            return [self._transform]
+
+        def predict_one(self):
+            """
+            Called to ask this displayable to call the callback with all
+            the images it may want to load.
+            """
+            if self._enable_events:
+                self.update_mouse_pos()
+            self._container._update_offsets()
+            self.update_offsets()
+
+
+    @store.mas_utils.deprecated(use_instead="ParallaxSprite")
+    class ParallaxBackground(renpy.display.core.Displayable):
```

Comment:
did this class exist before? Why would we need to deprecate instead of remove?

Reply:
Forgot to delete, used to have 2 classes.